docroot/modules/humsci/hs_migrate/src/HsMigrateBatchExecutable.php

```diff
@@ -0,0 +1,132 @@
+ *
+ * The primary object of this class is entirely to change the batch_limit in an
+ * effort to import 15 items on each batch execution.
+ *
+ * @package Drupal\hs_migrate
+ */
+class HsMigrateBatchExecutable extends MigrateBatchExecutable {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function batchImport() {
+    // Create the batch operations for each migration that needs to be executed.
+    // This includes the migration for this executable, but also the dependent
+    // migrations.
+    $operations = $this->batchOperations([$this->migration], 'import', [
+      'limit' => $this->itemLimit,
+      'update' => $this->updateExistingRows,
+      'force' => $this->checkDependencies,
+    ]);
+
+    if (count($operations) > 0) {
+      $batch = [
+        'operations' => $operations,
+        'title' => t('Migrating %migrate', ['%migrate' => $this->migration->label()]),
+        'init_message' => t('Start migrating %migrate', ['%migrate' => $this->migration->label()]),
+        'progress_message' => t('Migrating %migrate', ['%migrate' => $this->migration->label()]),
+        'error_message' => t('An error occurred while migrating %migrate.', ['%migrate' => $this->migration->label()]),
+        'finished' => '\Drupal\hs_migrate\HsMigrateBatchExecutable::batchFinishedImport',
+      ];
+
+      batch_set($batch);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function batchOperations(array $migrations, $operation, array $options = []) {
+    $operations = parent::batchOperations($migrations, $operation, $options);
+    foreach ($operations as &$operation) {
+      // Change the operation to use this class instead of the parent.
+      $operation[0] = [self::class, 'batchProcessImport'];
+    }
+    return $operations;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function batchProcessImport($migration_id, array $options, array &$context) {
+    if (empty($context['sandbox'])) {
+      $context['finished'] = 0;
+      $context['sandbox'] = [];
+      $context['sandbox']['total'] = 0;
+      $context['sandbox']['counter'] = 0;
+      $context['sandbox']['batch_limit'] = 0;
+      $context['sandbox']['operation'] = HsMigrateBatchExecutable::BATCH_IMPORT;
+    }
+
+    // Prepare the migration executable.
+    $message = new MigrateMessage();
+    /** @var \Drupal\migrate\Plugin\MigrationInterface $migration */
+    $migration = \Drupal::getContainer()
+      ->get('plugin.manager.migration')
+      ->createInstance($migration_id);
+    $executable = new HsMigrateBatchExecutable($migration, $message, $options);
+
+    if (empty($context['sandbox']['total'])) {
+      $context['sandbox']['total'] = $executable->getSource()->count();
+
+      // THIS is the only change from the parent class. Allow 15 items to be
+      // imported on each batch execution. The parent split the total items
+      // into 100 executions which doesn't really do anything helpful.
+      $context['sandbox']['batch_limit'] = 15;
```

Comment:
Can this be a configuration item, please? Seems like this could/would benefit from having a flexible batch limit if we ever move to different hardware.

Reply:
yeah. it'd be nice if migrate plus module would do that. i think they were lazy with it.